RequestContext::connection is unused

https://github.com/kiwix/kiwix-tools/blob/f99b6965e99f73f41de73ea32aed478888db7275/src/server/request_context.h#L87

This is currently set in the constructor:

https://github.com/kiwix/kiwix-tools/blob/f99b6965e99f73f41de73ea32aed478888db7275/src/server/request_context.cpp#L33

But it's not actually used anywhere. This reference is to the constructor argument, not the field:

https://github.com/kiwix/kiwix-tools/blob/f99b6965e99f73f41de73ea32aed478888db7275/src/server/request_context.cpp#L64

This causes Clang to throw a fit with warnings-as-errors:
```
../src/server/request_context.h:88:28: error: private field 'connection'
is not used [-Werror,-Wunused-private-field]
    struct MHD_Connection* connection;
                           ^
```

Not sure what your policy on such warnings is. It's clearly not a problem as such, but I'm not sure if there's some intent to use the field in the future or not. If not, it might be better to get rid of it in any case? I'll prepare a PR for removal in case you decide to go ahead with that.

--- a/src/server/request_context.cpp
+++ b/src/server/request_context.cpp
@@ -30,7 +30,6 @@ RequestContext::RequestContext(struct MHD_Connection* connection,
                                const std::string& _url,
                                const std::string& method,
                                const std::string& version) :
-  connection(connection),
   full_url(_url),
   url(_url),
   valid_url(true),

--- a/src/server/request_context.h
+++ b/src/server/request_context.h
@@ -85,7 +85,6 @@ class RequestContext {
     int httpResponseCode;
 
   private:
-    struct MHD_Connection* connection;
     std::string full_url;
     std::string url;
     bool valid_url;

